File: data/readable_format/fortran/file1820.f90
SUBROUTINE YA0013(IDA,nf1,nf5,nf6)
  INTEGER, INTENT(INOUT) :: IDA(9)
  INTEGER, INTENT(IN) :: nf1, nf5, nf6
  INTEGER :: i
  ! Initialize the array to 1
  DO i = 1, SIZE(IDA)
    IDA(i) = 1
  END DO
  ! Perform bit manipulation
  CALL MVBITS(IDA(nf5), 0, 1, IDA(nf6), 2)
END SUBROUTINE YA0013

PROGRAM TestYA0013
  INTEGER :: ida(9)
  INTEGER :: i
  CALL YA0013(ida,1,5,6)
  
  ! Check results and print the array
  DO i = 1, SIZE(IDA)
    PRINT *, 'IDA(', i, ') = ', IDA(i)
  END DO
END PROGRAM TestYA0013

File: data/readable_format/cpp/file1820.cpp
#include <iostream>
#include <vector>
#include <cassert>

void mvbits(int& from, int frompos, int len, int& to, int topos) {
    int mask = ((1 << len) - 1);
    int bits = (from >> frompos) & mask;
    to &= ~(mask << topos);
    to |= (bits << topos);
}

void ya0013(std::vector<int>& ida, int nf1, int nf5, int nf6) {
    std::fill(ida.begin(), ida.end(), 1);
    mvbits(ida[nf5 - 1], 0, 1, ida[nf6 - 1], 2);
}

void testYA0013() {
    std::vector<int> ida(9, 0); 
    ya0013(ida, 1, 5, 6);

    for(int i = 0; i < ida.size(); i++) {
        std::cout << "ida[" << i << "] = " << ida[i] << std::endl;
    }
}

int main() {
    testYA0013();
    return 0;
}